launch/model_endpoint.py

```diff
@@ -70,15 +71,45 @@ class EndpointResponse:
         Specifically, one can json.loads() the value of result to get the original python object back.
     """
 
-    def __init__(self, status, result_url, result):
+    def __init__(self, client, status, result_url, result):
+        self.client = client
         self.status = status
         self.result_url = result_url
         self.result = result
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"status: {self.status}, result: {self.result}, result_url: {self.result_url}"
 
 
+class EndpointResponseFuture:
+    def __init__(self, client, async_task_id: str):
+        self.client = client
+        self.async_task_id = async_task_id
+
+    def get(self) -> EndpointResponse:
+        while True:
+            async_response = self.client.get_async_response(self.async_task_id)
+            if async_response["state"] == "PENDING":
+                time.sleep(2)
```

Comment:
Should we do `async-await` stuff here? Or is that too complicated

Reply:
I think this is fine for now. Looks like the Nucleus `sleep_until_complete` function does a blocking poll too.

IMO generally if we are doing async `await`, we're probably using an entirely `asyncio`-ified version of this client library, or at least a segment of the whole client (kinda like how `boto3` has `aioboto3`).